tests/datasets/test_coco.py

```diff
@@ -0,0 +1,46 @@
+import os
+import tempfile
+from pathlib import Path
+from unittest.mock import patch
+
+from pytest import raises
+
+from datasetinsights.datasets.coco import CocoDetection
+from datasetinsights.datasets.exceptions import DatasetNotFoundError
+
+
+def test__is_dataset_files_present():
+    with tempfile.TemporaryDirectory() as tmp:
+        with open(os.path.join(tmp, "coco.json"), "x"):
+            with open(os.path.join(tmp, "coco.jpg"), "x"):
+                assert CocoDetection._is_dataset_files_present(tmp)
+
+    with tempfile.TemporaryDirectory() as tmp:
+        assert not CocoDetection._is_dataset_files_present(tmp)
+
+
+@patch("datasetinsights.datasets.CocoDetection._unarchive_data")
+def test__preprocess_dataset(mock_unarchive):
```

Comment:
[nit]: Two '_'.

Reply:
Since it's a private method (original name is `_preprocess_dataset`), so it should be two '_'

Reply:
Is this the naming convention for private method unit test?

Reply:
If the method name starts with '_', we should include the whole method name in a unit test. For example, `test__add_single_bbox_on_image_crop_label` in `test_visual.py`

Reply:
Otherwise, we may confuse people if we have other method named "preprocess_dataset".